### PR TITLE
I/O module documentation

### PR DESCRIPTION
**Pull request by h-2**

### CODE REVIEW

File: include/seqan3/io/all.hpp [hunk 1]
```diff
@@ -12,45 +12,90 @@
...
- * linked against your program (if you use CMake, this should happen automatically).
+ * | **File**                      | **Formats**                                                                                                  |
+ * |:------------------------------|:-------------------------------------------------------------------------------------------------------------|
+ * | seqan3::alignment_file_input  | seqan3::format_sam, seqan3::format_bam                                                                       |
+ * | seqan3::alignment_file_output | seqan3::format_sam, seqan3::format_bam                                                                       |
+ * | seqan3::sequence_file_input   | seqan3::format_embl, seqan3::format_fasta, seqan3::format_fastq, seqan3::format_genbank, seqan3::format_sam  |
+ * | seqan3::sequence_file_output  | seqan3::format_embl, seqan3::format_fasta, seqan3::format_fastq, seqan3::format_genbank, seqan3::format_sam  |
+ * | seqan3::structure_file_in  | seqan3::format_vienna                                                                                        |
```
Comment (eseiler) [inline include/seqan3/io/all.hpp]: alignment?

File: include/seqan3/io/all.hpp [hunk 2]
```diff
@@ -12,45 +12,90 @@
...
+ * links for more information.
+ *
+ * ## Records and fields
+ *
+ * The main file interface that SeqAn offers is *record-based*, i.e. every file conceptionally is a range of
+ * records. And each record in turn behaves as a tuple of fields.
+ *
+ * The record type of all files is based on seqan3::record, but the composition of fields is different **per file**.
+ *
+ * Maybe a little more concrete:
```
Comment (eseiler) [inline include/seqan3/io/all.hpp]:
```suggestion
 * In particular this means:
```
Or something similar, it sounds quite colloquial as it is.

File: include/seqan3/io/all.hpp (hunk 1 above)
Comment (h-2) [reply include/seqan3/io/all.hpp]: This was intentional, awaiting #1057

File: include/seqan3/io/all.hpp (hunk 2 above)
Comment (h-2) [reply include/seqan3/io/all.hpp]: Ok. I have another documentation PR in the pipeline and will update this inside their so we don't have to wait so much for travis, ok?

**Pull request merged**